mne/decoding/receptive_field.py

```diff
@@ -149,20 +152,30 @@ def fit(self, X, y):
             raise ValueError('scoring must be one of %s, got'
                              '%s ' % (sorted(_SCORERS.keys()), self.scoring))
         from sklearn.base import is_regressor, clone
-        X, y = self._check_dimensions(X, y)
+        X, y, _, self._y_dim = self._check_dimensions(X, y)
 
         # Initialize delays
         self.delays_ = _times_to_delays(self.tmin, self.tmax, self.sfreq)
 
         # Define the slice that we should use in the middle
-        self.keep_samples_ = _delays_to_slice(self.delays_)
+        self.valid_samples_ = _delays_to_slice(self.delays_)
 
         if isinstance(self.estimator, numbers.Real):
+            if self.fit_intercept is None:
+                self.fit_intercept = True
             estimator = TimeDelayingRidge(self.tmin, self.tmax, self.sfreq,
                                           alpha=self.estimator,
                                           fit_intercept=self.fit_intercept)
         elif is_regressor(self.estimator):
             estimator = clone(self.estimator)
+            if self.fit_intercept is not None and \
```

Comment:
so here we could just do `if self.fit_intercept is not None: "raise the error"` yeah?